### PR TITLE
Add contraction path cache in batch_tensor_contraction, support cotengra

### PR DESCRIPTION
**Pull request by LinjianMa**

### CODE REVIEW

File: src/ITensorAutoHOOT/batch_contraction.jl [hunk 1]
```diff
@@ -15,17 +15,43 @@ struct Executor
   feed_dict::Dict
 end
 
+struct NetworkCache
```
Comment (mtfishman) [inline src/ITensorAutoHOOT/batch_contraction.jl:18]: Could you explain what kinds of objects this stores?

For example, does it store a set of optimal einsum expressions, or a set of tensor networks (networks of ITensors)?

Comment (LinjianMa) [reply src/ITensorAutoHOOT/batch_contraction.jl:21]: Sure, just added some comments for this struct. NetworkCache.net_sums stores lists of AutoHOOT nodes (einsum expressions), and NetworkCache.index_dict maps each AutoHOOT node to the index of the ITensor tensor in the input array.

Comment (mtfishman) [reply src/ITensorAutoHOOT/batch_contraction.jl:21]: Thanks!

**Comment by mtfishman** [comment]
Looks good, this will be very helpful.

One thing we could try is moving the set of commands in the CI for installing `AutoHOOT` and `cotengra` into a file `deps/build.jl`, which Julia will run as a build step when the `ITensorNetworkAD` gets installed. Similar to the discussions here:
https://discourse.julialang.org/t/pycall-pre-installing-a-python-package-required-by-a-julia-package/3316
https://gist.github.com/Luthaf/368a23981c8ec095c3eb

This would make it easier for people to test out the package. We can leave that for a future issue.

**Review by mtfishman — APPROVED** [review]

**Pull request merged**